test/bot_test.exs

```diff
@@ -34,4 +35,21 @@ defmodule Slack.BotTest do
     assert slack.users    == %{"123" => %{id: "123"}}
     assert slack.ims      == %{"123" => %{id: "123"}}
   end
+
+  defmodule Slack.Rtm.Stub do
+    def start(_token) do
+      {:ok, %{url: "http://www.example.com"}}
+    end
+  end
+
+  defmodule Slack.WebsocketClient do
+    def start_link(_url, _module, _state, _options) do
+      {:ok, self()}
+    end
+  end
+
+  test "can configure the RTM module" do
+    Application.put_env(:slack, :rtm_module, Slack.Rtm.Stub)
+    assert {:ok, _pid} = :"Elixir.Slack.Bot".start_link(Bot, %{}, "token", %{client: Slack.WebsocketClient})
```

Comment:
curious is theres a reason why you're not using just `Slack.Bot` over `:"Elixir.Slack.Bot"`